quantecon/markov/approximation.py

```diff
@@ -212,3 +214,188 @@ def _fill_tauchen(x, P, n, rho, sigma, half_step):
+
+    For a mathematical derivation check *Finite-State Approximation Of
+    VAR Processes:  A Simulation Approach* by Stephanie Schmitt-Grohé and
+    Martín Uribe, July 11, 2010.
+
+    This code was adapted from Schmitt-Grohé and Uribe's original MATLAB code.
+
+    Parameters
+    ----------
+    A : array_like(float)
+        An m x m matrix containing the process' autocorrelation parameters
+    Omega : array_like(float)
+        An m x m variance-covariance matrix
+    grid_sizes : array_like(int) or None
+        An m-vector containing the number of grid points in the discretization
+        of each dimension of x_t. If grid_sizes is None, then grid_sizes is
+        set to (10, ..., 10).
+    std_devs : float
+        The number of standard deviations the grid should stretch in each
+        dimension, where standard deviations are measured under the stationary
+        distribution.
+    sim_length : int
+        The the length of the simulated time series (default,  1_000_000).
+    burn_in : int
+        The number of burn-in draws from the simulated series (default,
+        100_000).
+    row_major : bool
+        If True then return form the cartesian product state grid using row
+        major ordering.  Otherwise use column major.
+
+    Returns
+    -------
+    mc : MarkovChain
+        An instance of the MarkovChain class that stores the transition
+        matrix and state values returned by the discretization method
+
+
+    Notes
+    -----
+        The code presently assumes normal shocks but normality is not required
+        for the algorithm to work. The draws from the multivariate standard
+        normal generator can be replaced by any other random number generator
+        with mean 0 and unit standard deviation.
+
+
+    Example
+    -------
+
+        This example discretizes the stochastic process used to calibrate
+        the economic model included in ``Downward Nominal Wage Rigidity,
+        Currency Pegs, and Involuntary Unemployment'' by Stephanie
+        Schmitt-Grohé and Martín Uribe, Journal of Political Economy 124,
+        October 2016, 1466-1514.
+
+            A     = np.array([[0.7901, -1.3570],
+                              [-0.0104, 0.8638]])
+            Omega = np.array([[0.0012346, -0.0000776],
+                              [-0.0000776, 0.0000401]])
+            grid_sizes = np.array([21, 11])
+            mc = discrete_var(A, Omega, grid_sizes,
+                              sim_length=1_000_000, burn_in = 100_000)
+    """
+
+    m = len(A)   # The number of dimensions of the original state x_t
+    default_grid_size = 10
+
+    if grid_sizes is None:
+        # Set the size of every grid to default_grid_size
+        grid_sizes = np.full(m, default_grid_size)
+
+    n = grid_sizes.prod()  # Size of the discretized state
+
+    # Compute stationary variance-covariance matrix of AR process and use
+    # it to obtain grid bounds.
+    Sigma = qme.solve_discrete_lyapunov(A, Omega)
```

Comment:
Shouldn't it be
```suggestion
    Sigma = qme.solve_discrete_lyapunov(A, Omega @ Omega.T)
```
?